### PR TITLE
convert multichannel sfx to mono

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
We'll still have to handle the sndfile < 1.1.0 case gracefully. Sure, we'll lose MP3 support with this old version, but everything else is supposed to work as usual.

File: src/i_sndfile.c [hunk 1]
```diff
@@ -108,6 +108,80 @@ static SF_VIRTUAL_IO sfvio =
     sfvio_tell
 };
 
+sf_count_t sfx_mix_mono_read_float(SNDFILE *file, float *data, sf_count_t datalen)
```
Comment (fabiangreffrath) [inline src/i_sndfile.c:111]: static?

**Comment by rfomin** [comment]
> We'll still have to handle the sndfile < 1.1.0 case gracefully. Sure, we'll lose MP3 support with this old version, but everything else is supposed to work as usual.

There will be an error message and silence if the user tries to play an MP3 with sndfile < 1.1.0

**Comment by fabiangreffrath** [comment]
> There will be an error message and silence if the user tries to play an MP3 with sndfile < 1.1.0

Sounds reasonable. 🤷

File: cmake/FindSndFile.cmake [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
-#  SNDFILE_INCLUDE_DIRS - the SndFile include directory
-#  SNDFILE_LIBRARIES - Link these to use SndFile
-#
-#  Copyright © 2006  Wengo
-#  Copyright © 2009 Guillaume Martres
-#
-#  Redistribution and use is allowed according to the terms of the New
-#  BSD license.
-#  For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+# Variables defined:
```
Comment (fabiangreffrath) [inline cmake/FindSndFile.cmake:1]: Where is this file from? Copyright?

**Review by fabiangreffrath — APPROVED** [review]

File: cmake/FindSndFile.cmake (hunk 2 above)
Comment (rfomin) [reply cmake/FindSndFile.cmake:1]: It's from libsamplerate repository, no copyright. https://github.com/libsndfile/libsamplerate/blob/master/cmake/FindSndFile.cmake

Comment (fabiangreffrath) [reply cmake/FindSndFile.cmake:1]: Alright, could you leave that link there, please?

**Pull request merged**